neet/interfaces.py

```diff
@@ -112,13 +114,18 @@ def neighbors(net,index=None,direction='both',**kwargs):
     docstrings for more details and basic use examples.
 
     """
+    if direction not in ('in', 'out', 'both'):
+        raise ValueError('direction must be "in", "out" or "both"')
 
-    if net.__class__.__name__ == 'ECA':
+    neighbor_types = {'in': net.neighbors_in,
+                      'out': net.neighbors_out,
+                      'both': net.neighbors}
 
+    if net.__class__.__name__ == 'ECA':
         if 'size' not in kwargs:
             raise AttributeError("A `size` kwarg is required for returning an ECA's neighbors")
         else:
-            return net.neighbors(kwargs['size'],index=index,direction=direction)
+            return neighbor_types[direction](index, size=kwargs['size'])
```

Comment:
The way this is written, i think index needs to be a required argument, right?

Reply:
Yeah, I removed the default `None` from the function definition.